woocommerce/woocommerce-android
# Sync addons order product data

Summary
==========
## ⚠️This PR may look enormous but 70% of it are test fixtures and infrastructure like JSON files, don't be afraid 😬 

Fixes issue #4407 by creating a Repository and ViewModel classes responsible for fetching, parsing, and mapping the product add-ons contained inside an Order. 

As discussed in the past (p91TBi-4MK-p2), right now we don't have an effective way of connecting the Order Add-ons data with the Product ones since the Ordered Add-ons are set as Attributes inside a given Order Item. 

As a consequence, we need to do name and label matching to be able to identify which Add-ons were ordered. A terrible workaround, but it's what we have since there's no direct API for this plugin to make it easier to access the required data.

How to Test
==========
The only way to effectively test this PR is through unit tests since there's nothing getting delivered to a View right now. Since what we're doing here is a workaround, I decided to be very aggressive with the unit test coverage, doing lots of integrations between the data models, repository, and the ViewModel to be sure we're not breaking anything and the string matching between Order Attributes and Product Add-ons are done as expected. 

My recommendation is to test this feature by exploring the scenarios of the Unit tests, see if there are more scenarios we could add to improve our coverage, and of course, making sure all tests are green.

Update release notes:
